lib/avtp_pipeline/openavb_common/mrp_client.c

```diff
@@ -203,11 +209,37 @@ int process_mrp_msg(char *buf, int buflen)
 		goto next_line;
 		break;
 	case 'T':
-
-		/* as simple_talker we don't care about other talkers */
 		i = k;
+		while (buf[i] != 'S')
+			i++;
+		// skip S=
+		i += 2;
+		for (j = 0; j < 8; j++) {
+			sscanf(&(buf[i + 2 * j]), "%02x", &id);
+			recovered_streamid[j] = (unsigned char)id;
+		}
+		while (buf[i] != 'A')
+			i++;
+		// skip A=
+		i += 2;
+		for (j = 0; j < 6; j++) {
+			sscanf(&(buf[i + 2 * j]), "%02x", &id);
+			dest_addr[j] = (unsigned char)id;
+		}
+		i+= j*2 + 1;
+
+		sscanf(&(buf[i]), "V=%d,Z=%d,I=%d,P=%d,L=%d",
+		       &vid,
+		       &max_frame_size,
+		       &max_interval_frames,
+		       &priority_and_rank,
+		       &latency);
+
+		mrp_register_cb(recovered_streamid, 0, dest_addr, max_frame_size, max_interval_frames, vid, latency);
+
```

Comment:
If I were messing with messages from mrpd, I would use this code https://github.com/AVnu/Open-AVB/blob/master/examples/mrp_client/mrpdhelper.h